python/tests/e2e/conftest.py

```diff
@@ -47,13 +54,15 @@ class VCRConfig(TypedDict):
     Common options:
     - 'method': HTTP method (GET, POST, etc.)
     - 'uri': Request URI/URL
-    - 'body': Request body content
+    - 'body': Request body content (use 'raw_body' for exact binary matching)
     - 'headers': Request headers
+    - 'scheme', 'host', 'port', 'path', 'query': URL components
     """
 
     filter_headers: list[str]
     """Headers to filter out from recordings for security/privacy.
 
+    DEPRECATED: Use before_record_request instead for better control.
```

Comment:
If we're dropping usage of this, let's just drop it from the VCRConfig as a clean refactor? (I would suggest refactoring this file as a separate PR at the bottom of the stack, so we can merge it into trunk quickly separately from the review cycle on the mirrored clients)